src/tls_relation.py

```diff
@@ -12,23 +12,31 @@
     CertificateAvailableEvent,
     CertificateExpiringEvent,
     CertificateInvalidatedEvent,
+    TLSCertificatesRequiresV2,
     generate_csr,
     generate_private_key,
 )
-from ops.charm import CharmBase
+from cryptography.hazmat.primitives import serialization
 from ops.jujuversion import JujuVersion
-from ops.model import Application, Relation, SecretNotFoundError
+from ops.model import Application, Model, Relation, SecretNotFoundError
 
 from consts import PEER_RELATION_NAME, TLS_CERT
 
 
 class TLSRelationService:
     """TLS Relation service class."""
 
-    def __init__(self) -> None:
-        """Init method for the class."""
+    def __init__(self, model: Model, app: Application) -> None:
```

Comment:
Isn't app the same as model.app?

Reply:
Yes it is, let's say it's a first step of improvement compared to the fact that multiple methods so far were taking the whole charm (leading to a dependency with the charm in this module)

Currently I need the model and app for manage secrets, I'd like to find a solution for this case
model is also used to manage the relations but they might be passed to the constructor